### DISCUSSION TITLE
Spacing transform does not update transformed image meta dictionary?

### DISCUSSION THREAD

**Discussion by agaldran** [Project-MONAI/MONAI#6831] [Q&A] [answered]
Hello,

Sorry about the naive question, I am trying to wrap my head around transforms and meta_dicts. Right now I have the following operations:
```
import monai.transforms as t

data_dict = {'image': 'data/ct/images/.whatever.nii.gz'}
loader = t.LoadImaged(keys=('image'), image_only=False)
loaded_data_dict = loader(data_dict)
print(loaded_data_dict['image'].shape, loaded_data_dict['image_meta_dict']['pixdim'][1:4])
```
which returns a shape and voxel sizes of `[281, 313, 182]` and `(0.50,0.50,0.75)` respectively. Now, if I run:
```
spacing = t.Spacingd(keys=['image'], pixdim=(0.50, 0.50, 1.5), mode=('bilinear')) 
spaced_data_dict = spacing(loaded_data_dict)
print(spaced_data_dict['image'].shape, spaced_data_dict['image_meta_dict']['pixdim'][1:4])
```
I am expecting to see approx half the voxels in the Z direction, as the original spacing was 0.75 and I am asking for twice that spacing. For the other directions, nothing should be changed^* . This is what I get:
```
[281, 313, 92], (0.50,0.50,0.75)
```
So, although the number of voxels are as expected, the spacing remains the same. I cannot find any other key in the meta data dictionary that tells me what is the *current* spacing for this volume. Am I doing something wrong here?

^* not sure about this either but numbers seem to match. Is my understanding correct? It also happens that if I use (0.0, 0.0, 1.5) I get the same result as when I specify the original voxel sizes, is this expected behavior?

**Comment by KumoLiu** [comment, marked as answer]
Hi @agaldran, thanks for your interest here.
- First, `pixdim` should also be updated after `Spacing`, I think it's a bug here, so I will create an issue for it.
- You can also use `spaced_data_dict['image_meta_dict']["affine"]` to get the new affine which contains the pixdim information.
- When you use non-positive values, the transform will use the corresponding components of the original pixdim, which is computed from the `affine` matrix of the input image, so it will return the original voxel size, which is expected.
https://github.com/Project-MONAI/MONAI/blob/cb257d249adc8d5e8c7c586191a92dd6db2abc4c/monai/transforms/spatial/array.py#L362-L369

BTW, you can use `spaced_data_dict["image"].meta` instead of `spaced_data_dict['image_meta_dict']` to get meta information directly after monai v0.9.1.
https://github.com/Project-MONAI/MONAI/wiki/MetaTensor-guide
Hope it helps, thanks!

**Comment by agaldran** [comment]
That is very interesting thanks a lot! 

Expect me to be back in this Q&A in the next weeks, I am finally having some time to dig in MONAI code and I my hat is full of questions :)

Cheers,

Adrian
